File: old/dekaf-core/src/main/java/org/jetbrains/dekaf/core/BaseCommandRunner.java
package org.jetbrains.dekaf.core;

import org.jetbrains.dekaf.intermediate.IntegralIntermediateSeance;



/**
 * @author Leonid Bushuev from JetBrains
 */
public class BaseCommandRunner implements DBCommandRunner, BaseSeanceRunner {

  private final IntegralIntermediateSeance myInterSeance;

  protected BaseCommandRunner(final IntegralIntermediateSeance interSeance) {
    myInterSeance = interSeance;
  }

  @Override
  public DBCommandRunner withParams(final Object... params) {
    myInterSeance.setInParameters(params);
    return this;
  }

  @Override
  public DBCommandRunner run() {
    myInterSeance.execute();
    return this;
  }


  @Override
  public void close() {
    myInterSeance.close();
  }
}

File: dev/dekaf-main/src/impl/BaseCommandRunner.kt
package org.jetbrains.dekaf.impl

import org.jetbrains.dekaf.core.DBCommandRunner
import org.jetbrains.dekaf.core.TaskKind.TASK_COMMAND
import org.jetbrains.dekaf.inter.InterSeance
import org.jetbrains.dekaf.inter.InterTask


internal class BaseCommandRunner: BaseStatementRunner, DBCommandRunner {

    internal constructor(session: BaseSession, interSeance: InterSeance, text: String) : super(session, interSeance, text)

    override fun prepare() {
        val task = InterTask(TASK_COMMAND, text)
        interSeance.prepare(task)
        prepared = true
    }

    override fun withParams(vararg params: Any?): BaseCommandRunner {
        super.withParams(*params)
        return this
    }

    override fun run(): BaseCommandRunner {
        if (!prepared) prepare()
        interSeance.execute()
        return this
    }

    override val affectedRowsCount: Int
        get() = interSeance.affectedRowsCount
}